src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,444 @@
+                return [];
+            }
+
+            var indicators = [];
+
+            for (
+                var indicatorValue = 0;
+                indicatorValue <= setting.schema.max;
+                indicatorValue = parseFloat((indicatorValue + setting.schema.divisibleBy).toPrecision(2))
+            ) {
+                if (indicatorValue === 0) {
+                    indicators.push({
+                        indicatorValue: setting.schema.min, // value cannot be 0, instead use setting minimum value
+                        isSelected: setting.schema.min === setting.value,
+                        isRecommended: setting.schema.min === setting.schema["default"]
+                    });
+                } else {
+                    indicators.push({
+                        indicatorValue: indicatorValue, // what value to be applied when selected
+                        isSelected: indicatorValue === setting.value,
+                        isRecommended: indicatorValue === setting.schema["default"]
+                    });
+                }
+            }
+            return indicators;
+        }
+    };
+
+    /**
+     * Handler for a single indicator element.
+     *
+     * Each indicator element has three states: normal, selected and default.
+     * These three states are indicated using a custom html element
+     * attribute - "data-type". Depending on the state of this attribute, different
+     * styles are applied (refer to the CSS for more info).
+     */
+    fluid.defaults("gpii.qssWidget.mouse.indicator.presenter", {
+        gradeNames: ["fluid.viewComponent", "gpii.app.clickable"],
+
+        model: {
+            item: {
+                indicatorValue: null,
+                isSelected: null,
+                isRecommended: null
+            }
+        },
+
+        stateAttribute: {
+            attrName: "data-type",
+            values: {
+                selected: "selected",
+                recommended: "recommended"
+            }
+        },
+
+        modelListeners: {
+            item: {
+                funcName: "gpii.qssWidget.mouse.indicator.updateState",
+                args: [
+                    "{that}.container",
+                    "{that}.options.stateAttribute",
+                    "{that}.model.item"
+                ]
+            }
+        },
+
+        listeners: {
+            onClicked: {
+                func: "{gpii.qssWidget.mouse.indicators}.events.onIndicatorClicked.fire",
+                args: "{that}.model.item.indicatorValue"
+            }
+        }
+    });
+
+    /**
+     * Alters the custom element attribute in order to change the styles applied to it.
+     * @param {jQuery} indicatorContainer - The container for the indicator element
+     * @param {Object} stateAttribute - Options for the state defining attribute
+     * @param {Object} indicatorData - The condition data for the element
```

Comment:
Please add @typedef structures for these two argumens

Reply:
Actually this whole function can be axed as identical to gpii.qssWidget.baseStepper.indicator.updateState - please check for other duplications